src/warnings.ts

```diff
@@ -0,0 +1,156 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as chalk from 'chalk';
+import * as fs from 'fs';
+import * as path from 'path';
+
+import {SourceRange} from './ast/source-range';
+import {Severity, Warning} from './editor-service';
+
+export enum Verbosity {
+  OneLine,
+  Full,
+}
+
+export interface PrinterOptions {
+  basedir: string;
+  verbosity?: Verbosity;
+  color?: boolean;
+}
+export interface FilterOptions {
+  warningCodesToIgnore?: Set<string>;
+  minimumSeverity: Severity;
+}
+
+const defaultFilterOptions: FilterOptions = {
+  warningCodesToIgnore: new Set(),
+  minimumSeverity: Severity.INFO
+};
+
+export class WarningFilterer {
+  constructor(private _options: FilterOptions) {
+    this._options = Object.assign({}, defaultFilterOptions, this._options);
+  }
+
+  shouldIgnore(warning: Warning) {
+    if (this._options.warningCodesToIgnore.has(warning.code)) {
+      return true;
+    }
+    if (warning.severity > this._options.minimumSeverity) {
+      return true;
+    }
+    return false;
+  }
+}
+
+const defaultPrinterOptions = {
+  verbosity: Verbosity.Full,
+  color: true
+};
+
+export class WarningPrinter {
+  _chalk: typeof chalk;
+  constructor(
+      private _outStream: NodeJS.WritableStream,
+      private _options?: PrinterOptions) {
+    this._options = Object.assign({}, defaultPrinterOptions, _options);
+    this._chalk =
+        new (chalk.constructor as any)({enabled: this._options.color});
```

Comment:
What's wrong here that we need to use `any`? Is this something that can be fixed in the @types declaration?